components/brave_rewards/browser/extension_rewards_notification_service_observer.h

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_REWARDS_BROWSER_REWARDS_NOTIFICATION_SERVICE_OBSERVER_EXTENSIONS_IMPL_
+#define BRAVE_COMPONENTS_BRAVE_REWARDS_BROWSER_REWARDS_NOTIFICATION_SERVICE_OBSERVER_EXTENSIONS_IMPL_
+
+#include <memory>
```

Comment:
I don't think you need memory or string here?

Reply:
headers should try to have the minimum possible includes. Use forward decls where possible and prefer to use includes in the .cc files. It reduces the size of rebuilds when header files change

Reply:
obviously that doesn't apply to memory or string, but just cleaner to only include what you are using

Reply:
Fixed.